core/src/main/java/org/elasticsearch/search/aggregations/metrics/geocentroid/InternalGeoCentroid.java

```diff
@@ -145,14 +146,27 @@ public Object getProperty(List<String> path) {
     }
 
     static class Fields {
-        static final String CENTROID = "location";
+        static final ParseField CENTROID = new ParseField("location");
+        static final ParseField COUNT = new ParseField("count");
+        static final ParseField CENTROID_LAT = new ParseField("lat");
+        static final ParseField CENTROID_LON = new ParseField("lon");
     }
 
     @Override
     public XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
+        return renderXContent(builder, params, centroid, count);
+    }
+
+    static XContentBuilder renderXContent(XContentBuilder builder, Params params, GeoPoint centroid, long count) throws IOException {
```

Comment:
Good point about the parse fields. Let's not worry too much for now. We can't be consistent if we don't enforce these constraints. Let's share these simple things for now and one day we will change that when we need to.